src/live/test-runner.js

```diff
@@ -0,0 +1,176 @@
+import Promise from 'pinkie';
+import { noop } from 'lodash';
+import LiveModeTestRunController from './test-run-controller';
+import Controller from './controller';
+import Runner from '../runner';
+import LiveModeBootstrapper from './bootstrapper';
+import parseFileList from '../utils/parse-file-list';
+import { GeneralError } from '../errors/runtime';
+import MESSAGE from '../errors/runtime/message';
+
+class LiveModeRunner extends Runner {
+    constructor (proxy, browserConnectionGateway, options) {
+        super(proxy, browserConnectionGateway, options);
+
+        /* EVENTS */
+        this.TEST_RUN_STARTED            = 'test-run-started';
+        this.TEST_RUN_DONE_EVENT         = 'test-run-done';
+        this.REQUIRED_MODULE_FOUND_EVENT = 'require-module-found';
+
+        this.stopping            = false;
+        this.tcRunnerTaskPromise = null;
+        this.stopInfiniteWaiting = noop;
+        this.preventRunCall      = false;
+
+        this.testRunController = new LiveModeTestRunController();
+
+        this.testRunController.on(this.testRunController.RUN_STARTED_EVENT, () => this.emit(this.TEST_RUN_STARTED, {}));
+
+        this.embeddingOptions({
+            TestRunCtor: this.testRunController.TestRunCtor,
+            assets:      []
+        });
+
+        this.controller = this._createController();
+    }
+
+    runTests (isFirstRun = false) {
+        let runError = null;
+
+        return this._finishPreviousTestRuns()
+            .then(() => {
+                return this._validateRunnableConfiguration(isFirstRun);
+            })
+            .then(() => {
+                this.tcRunnerTaskPromise = super.run(this.opts);
+
+                return this.tcRunnerTaskPromise;
+            })
+            .catch(err => {
+                this.setBootstrappingError(null);
+
+                runError = err;
+            })
+            .then(() => {
+                this.tcRunnerTaskPromise = null;
+
+                this.emit(this.TEST_RUN_DONE_EVENT, { err: runError });
+            });
+    }
+
+    _createRunnableConfiguration () {
```

Comment:
```js
async _createRunnableConfiguration () {
    if (this.liveConfigurationCache)
	 return this.liveConfigurationCache;

    const configuration  = await super._createRunnableConfiguration();

    this.liveConfigurationCache = configuration;

    return configuration;
}
```

Reply:
there is an error with using `super` inside an `async` function

Reply:
Ok